Initial Codebase for shopping cart assignment

jackvik opened the pull request.

jackvik committed:
Codebase for shopping cart assignment

jackvik committed:
Added the partial feature in Home Page

Added the display of category cards on the home page

jackvik committed:
Added the ProductCard and Product Listing page

Added the redux code to fetch all the products and fetch all the categories. Also added the product card view and listed the product cards. all implementation doing in mobile view.

jackvik committed:
Added Cart Functionality

jackvik committed:
Update cart feature

Update code for total price

ankush29 commented on dist/index.html:
```diff
@@ -0,0 +1,14 @@
+    <script defer src="index_bundle.js"></script></head>
+    <body>
+        <div id="root"></div>
+        <script src="app.js"></script>
```
why script tag is here?

ankush29 commented on dist/index_bundle.js:
```diff
@@ -0,0 +1,32 @@
+/******/ 	__webpack_modules__["./src/index.js"]();
+/******/ 	
+/******/ })()
+;
```
this should be a part of gitignore

ankush29 commented on package-lock.json:
```diff
@@ -0,0 +1,9452 @@
+{
```
again git ignore

ankush29 commented on package.json:
```diff
@@ -3,13 +3,48 @@
+    "start": "webpack serve --open",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "canned -p 5000 ./server"
+    "server": "canned -p 5000 ./server",
```
concurrently  run both commands server and start

ankush29 commented on package.json:
```diff
@@ -3,13 +3,48 @@
-    "start": "canned -p 5000 ./server"
+    "server": "canned -p 5000 ./server",
+    "create": "webpack"
   },
```
where is build command

ankush29 commented on package.json:
```diff
@@ -3,13 +3,48 @@
   "dependencies": {
-    "canned": "^0.3.12"
+    "canned": "^0.3.12",
+    "clean-webpack-plugin": "^4.0.0-alpha.0",
```
differentiate dependency and devDependency, don't mix them

ankush29 commented on webpack.config.js:
```diff
@@ -0,0 +1,39 @@
+    plugins: [
+
+        new HtmlWebpackPlugin({
+            template: 'src/index.html'
```
it should be a part of public folder

ankush29 commented on src/App.js:
```diff
@@ -0,0 +1,47 @@
+    return (
+        <>
+             <Router>
+          <Header/>
```
indentation

ankush29 commented on src/App.js:
```diff
@@ -0,0 +1,47 @@
+                    <Redirect to="/home" />
+                </Route>
+                <Route exact path='/home'>
+                    <Home/>
```
use component prop or render props pattern

ankush29 commented on src/App.js:
```diff
@@ -0,0 +1,47 @@
+                        <Cart/>
+                </Route>
+                <Route exact path='/signin' component={SignInPage} />
+                <Route exact path='/register' component={RegisterPage} />
```
what about 404

ankush29 commented on src/actions/actionCreators.js:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  export function showModal(){
+    console.log("action called");
```
remove console

ankush29 commented on src/actions/actionCreators.js:
```diff
@@ -0,0 +1,43 @@
+      return getData(GET_BANNERS_API).then(json => {
+          dispatch({ type: GET_BANNERS, payload: json });
+        });
+    };
```
why are saving all response in store?

ankush29 commented on src/app.scss:
```diff
@@ -0,0 +1,260 @@
+a {
+    font-family: 'Ropa Sans', sans-serif;
+}
+  .categoryContainer{
```
class name and id's not should be in camel-case, use bem

ankush29 commented on src/app.scss:
```diff
@@ -0,0 +1,260 @@
+      margin-bottom: 2rem;
+      box-shadow: 0 13px 25px -23px #a5a5a5ad;
+    }
+    .productCard{
```
css should be break out in different component, every component should have it's own css

ankush29 replied:
folder structure is not looks good -> https://reactjs.org/docs/faq-structure.html

Grouping by features or routes

ankush29 commented on src/components/molecules/Cart.js:
```diff
@@ -0,0 +1,87 @@
+  const history = useHistory();
+
+ 
+  const data = useSelector(state => state.cart.cartList);
```
use meaningful variable name

ankush29 commented on src/components/molecules/Cart.js:
```diff
@@ -0,0 +1,87 @@
+        </button>
+      </div>
+      <div className="modal-body" >
+      {data.length ? (
```
divide section into two parts one for list and another for empty cart

ankush29 commented on src/components/molecules/CartItem.js:
```diff
@@ -0,0 +1,75 @@
+                    </button>
+
+                    <span className="cart-qty">{item.count}</span>
+                    <button
```
create common button component and follow children pattern

ankush29 commented on src/components/molecules/CartItem.js:
```diff
@@ -0,0 +1,75 @@
+
+                    <span>X</span>
+                    <span>
+                        ₹
```
indentation

ankush29 commented on src/components/molecules/CategoryCard.js:
```diff
@@ -0,0 +1,21 @@
+        </div>
+        <div className="cardContent">
+            <h3>{props.categoryName}</h3>
+            <br></br>
```
avoid br

ankush29 commented on src/components/molecules/CategoryFilter.js:
```diff
@@ -0,0 +1,8 @@
+import React,{useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+export default CategoryFilter = (props)=>{
```
not get it..remove unused code

ankush29 commented on src/components/molecules/Header.js:
```diff
@@ -0,0 +1,63 @@
+                    {showModalFlag?
+            <div className="cartContainer">
+              <Cart />
+            </div>:""}
```
use null

ankush29 commented on src/components/molecules/ProductCard.js:
```diff
@@ -0,0 +1,33 @@
+    const productObj = product;
+    const dispatch = useDispatch();
+    const buyNowClickHandler = (item) => {   
+        console.log(item);           
```
remove console

ankush29 commented on src/components/molecules/SigninForm.js:
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+class SigninForm extends React.Component{
```
why class component?

ankush29 commented on src/components/molecules/SigninForm.js:
```diff
@@ -0,0 +1,50 @@
+                    <input type="email" name="email" onChange={this.handleChange} id="email" required="required" />
+                </div>
+                <div className="group">
+                    <input type="password" name="password" onChange={this.handleChange} id="password" required="required" />
```
use label

ankush29 commented on src/components/pages/Home.js:
```diff
@@ -0,0 +1,25 @@
+    return (<>
+    
+            <div className="categoryContainer">
+            {categoriesList.categories?.map(a=>{return <CategoryCard key={a.key} categoryName={a.name} imgUrl={a.imageUrl} desc={a.description}/>})}
```
create separate category component and pass only required information to it

ankush29 commented on src/components/pages/ProductListingPage.js:
```diff
@@ -0,0 +1,59 @@
+    const [showDropdown,setShowDropdown] = useState(false);
+    const [displayList,setDisplayList] = useState([]);
+    const dispatch = useDispatch();
+    React.useEffect(()=>{
```
import useEffect and then use

ankush29 commented on src/components/pages/ProductListingPage.js:
```diff
@@ -0,0 +1,59 @@
+      dispatch(getProducts());
+
+    },[])
+    React.useEffect(() => {
```
again

ankush29 commented on src/components/pages/ProductListingPage.js:
```diff
@@ -0,0 +1,59 @@
+      setDisplayList(productsList);
+    }, [productsList]);
+
+    const categoriesDropdown = categoriesList.categories?.map((category) => {
```
move into different component

ankush29 commented on src/components/pages/ProductListingPage.js:
```diff
@@ -0,0 +1,59 @@
+    return (<>
+        <div className="categoriesDropdownHeader"  onClick={()=>setShowDropdown(true)} >{selectedCategory || "Select Category"}</div>
+        {showDropdown && <ul className="categoriesDropdown">
+            {categoriesDropdown}
```
make it tag </>

ankush29 commented on src/components/pages/RegisterPage.js:
```diff
@@ -0,0 +1,10 @@
+const RegisterPage =()=>{
+    return (<>
+            <p>RegisterPage</p>
+    
```
incomplete

jackvik committed:
Improvements

jackvik committed:
Few Changes to fix UI

jackvik committed:
Banner API and Cart Display Changes

jackvik committed:
Removed unused code

jackvik committed:
Added Footer

jackvik committed:
Added Not found route

jackvik committed:
Some fixes for css and aria label

jackvik committed:
Some Improvements

Removed the redux lofic for categories and banners and added a jest configuration

jackvik committed:
Minor Fixes

jackvik committed:
Minor Fixes

jackvik committed:
Removed Unused Redux Code

jackvik committed:
Fixed no found route

jackvik committed:
Few Changes for snapshot test

jackvik committed:
Update broken package.json

jackvik committed:
Changes for app not found error and cart height

jackvik committed:
Minor UI hanges

jackvik committed:
Product Card broken fix for Tab

jackvik committed:
Fixed Formatting

jackvik committed:
Fixed minor CSS change

jackvik committed:
Updated the UI Changes

jackvik committed:
Updated Some Changes

jackvik committed:
Updated gitignore

jackvik committed:
Updated Minor Changes

jackvik committed:
Removed unused CSS and fixed some bugs

jackvik committed:
Update gitignore

jackvik committed:
Updated Changes for Query Param and other optimization

jackvik committed:
Refactored Login/Register

jackvik committed:
Minor Changes

jackvik committed:
Removed Unused Files
